src/plugins/intellisense/index.tsx

```diff
@@ -0,0 +1,512 @@
+    this.view?.update();
+  }
+
+  // leave an intellisense menu and return to whatever expression
+  // you were previously in
+  leaveIntellisenseMenu() {
+    if (this.intellisenseReturnMQ) {
+      this.intellisenseReturnMQ?.focus();
+      this.latestMQ = this.intellisenseReturnMQ;
+    }
+
+    if (this.prevCursorPos && this.latestMQ) {
+      const mqRootBlock =
+        this.latestMQ.__controller.container.querySelector(
+          ".dcg-mq-root-block"
+        );
+
+      if (!mqRootBlock) return;
+
+      const mqRootBlockRect = mqRootBlock.getBoundingClientRect();
+
+      // simulate a click to get cursor in the right spot
+      const eventHandlerSettings = {
+        bubbles: true,
+        clientX:
+          this.prevCursorPos.x - mqRootBlock.scrollLeft + mqRootBlockRect.x,
+        clientY:
+          this.prevCursorPos.y - mqRootBlock.scrollTop + mqRootBlockRect.y,
+      };
+      this.latestMQ.__controller.container.dispatchEvent(
+        new MouseEvent("mousedown", eventHandlerSettings)
+      );
+      this.latestMQ.__controller.container.dispatchEvent(
+        new MouseEvent("mouseup", eventHandlerSettings)
+      );
+    }
+  }
+
+  // keep track of where the cursor is so we can return to it
+  // once we refocus the mathquill input
+  saveCursorState() {
+    const focusedmq = MathQuillView.getFocusedMathquill();
+    if (focusedmq) this.latestMQ = focusedmq;
+    if (
+      this.latestMQ &&
+      document.body.contains(this.latestMQ.__controller.cursor.cursorElement)
+    ) {
+      // get cursor pos relative to the top left of the mathquill's root element
+      const mqRootBlock =
+        this.latestMQ.__controller.container.querySelector(
+          ".dcg-mq-root-block"
+        );
+
+      if (!mqRootBlock) return;
+      const mqRootBlockRect = mqRootBlock.getBoundingClientRect();
+      const rect =
+        this.latestMQ.__controller.cursor.cursorElement.getBoundingClientRect();
+      this.prevCursorPos = {
+        x: rect.x + mqRootBlock.scrollLeft - mqRootBlockRect.x,
+        y: rect.y + mqRootBlock.scrollTop - mqRootBlockRect.y,
+      };
+    }
+  }
+
+  focusOutHandler = (e: FocusEvent) => {
+    if (
+      e.target !== intellisenseMountPoint &&
+      e.relatedTarget !== intellisenseMountPoint &&
+      e.relatedTarget !== null
+    ) {
+      this.canHaveIntellisense = false;
+      this.view?.update();
+    }
+  };
+
+  modifiedOverrideKeystrokeUnsubbers: (() => void)[] = [];
+
+  focusInHandler = () => {
+    setTimeout(() => {
+      const mqopts = Calc.focusedMathQuill?.mq?.__controller?.options;
+
+      // done because the monkeypatch has different a different this value
+      // eslint-disable-next-line @typescript-eslint/no-this-alias
+      const self = this;
+
+      if (mqopts && !(mqopts.overrideKeystroke as any).isMonkeypatchedIn) {
+        // monkeypatch in a function to wrap overrideKeystroke
+        const remove = attach<(key: string, evt: KeyboardEvent) => void>(
```

Comment:
This is some beautiful syntax (though why is it necessary? Can't TS infer based on the type of the second argument?)

Reply:
Weird... seems like it does work. Earlier it was thinking the function had no parameters. Might just be some kind of cache issue.

Reply:
Fixed